src/openloco/townmgr.cpp

```diff
@@ -140,6 +137,6 @@ namespace openloco::townmgr
             memset(&currTown.unk_158, 0, 20 * sizeof(uint16_t));
         }
 
-        ui::WindowManager::invalidate((ui::WindowType)0x61);
+        ui::WindowManager::invalidate((ui::WindowType)((uint8_t)ui::WindowType::town | (uint8_t)ui::WindowType::townList));
```

Comment:
I don't think this is correct. town = 33 (0x21) and townList = 32 (0x20), which does not *or* to 0x61.

As @marijnvdwerf suggested, it appears this is a combination of `ui::WindowType::town` and a flag 0x40 instead.

Reply:
I checked again and it seems the code just invalidates all town windows:
https://github.com/OpenRCT2/OpenLoco/blob/master/src/openloco/ui/WindowManager.cpp#L176
`0x61 & (1<<6) != 0`, so we calculate `0x61 & 0xBF = 0x21 = 33`, which is the WindowType for towns. Or am I missing something?

Reply:
`ui::window_flags::flag_6` isn't right. If I remember rightly there is a flag used in the original code to say whether it is also passing a window number as well as a class, and maybe another flag for whether it is also passing a widget to invalidate. We do not need to keep those flags as we have three overloads for the invalidate method.

Reply:
Ah, right, Sawyer used this flag to change function behaviour, but we're overloading the functions anyway. Should be fine to just have `ui::WindowType::town` then.

```diff
-        ui::WindowManager::invalidate((ui::WindowType)((uint8_t)ui::WindowType::town | (uint8_t)ui::WindowType::townList));
+        ui::WindowManager::invalidate(ui::WindowType::town);
```